hub/core/transform/transform_tensor.py

```diff
@@ -0,0 +1,59 @@
+from hub.core.sample import Sample  # type: ignore
```

Comment:
what are the differences between `TransformDatasetShard` and `TransformDatasetTensor`?

also, maybe we can drop the `Dataset` from both class names?

Reply:
they're literally basic version of Dataset and Tensor respectively, used for the sample_out argument. 
The underlying implementation does not involve any chunking/compression though, raw data is just stored.

Reply:
in that case, is it possible to use Dataset & Tensor?

if i'm a user, a likely thing i will do is print out the variables, and if i see Dataset/Tensor i will know exactly what they are.

Reply:
also it makes more sense to me to call these "View"s even if it's not as technically descriptive as "Shard"

Reply:
yeah but this isn't exactly a Dataset/Tensor, it's just a dummy class representing it. So a lot of the methods/attributes of Dataset won't work.

Reply:
What about TransformDataset and TransformTensor?

Reply:
yep that works well